crates/worker/src/partition/snapshot_producer.rs

```diff
@@ -69,6 +71,16 @@ impl SnapshotProducer {
             "Partition snapshot written"
         );
 
+        let previous_archived_snapshot_lsn = partition_store.get_archived_lsn().await?;
+        let mut tx = partition_store.transaction();
+        tx.put_archived_lsn(snapshot.min_applied_lsn).await;
+        tx.commit().await?;
```

Comment:
It was a suggestion somewhere along the line, but I don't think we _need_ it, and it might actually simplify life not to track it as an FSM variable. We can use the object store as the source of truth for this and load it asynchronously on start.

Reply:
Sounds like a good idea to not track it as a FSM variable then.

Reply:
Done!